mobile/ios/WirelessDebug/WirelessDebugger/WebSocketMessenger.swift

```diff
@@ -0,0 +1,113 @@
+//
+//  WebSocketMessenger.swift
+//  Wireless Debug iOS Test App
+//
+
+import UIKit
+import Foundation
+
+/// Handles opening a WebSocket connection and sending the necessary messages
+/// across that connection.
+class WebSocketMessenger {
+    var webSocket = WebSocket()
+    var logsToSend: [String] = []
+    var connectionRetries = 0
+    var verbose: Bool
+
+    /// Create a WebSocket connection to the given socket address and API Key.
+    ///
+    /// - Parameters:
+    ///   - socketAddress: the WebSocket address to connect to
+    ///   - apiKey: the API Key provided by the web interface
+    init(_ socketAddress: String, apiKey: String, verbose: Bool) {
+        self.verbose = verbose
+        // When the WebSocket connection opens, send a startSession message.
+        self.webSocket.event.open = {
+            self.log("WebSocket Connection to \(socketAddress) Opened")
+
+            self.send("startSession", data: [
+                "osType": "iOS",
+                "apiKey": apiKey,
+                "deviceName": UIDevice.current.name,
+                "appName": Bundle.main.infoDictionary!["CFBundleName"] as! String,
+            ])
+            self.connectionRetries = 0
+        }
+
+        // If the connection closes, try to reconnect 10 times.
+        self.webSocket.event.close = { code, reason, clean in
+            self.log("WebSocket Connection Closed \(reason)")
+
+            // Retry the connection 10 times with 2 seconds inbetween retries.
+            if self.connectionRetries < 10 {
+                DispatchQueue.main.asyncAfter(deadline: .now() + 2.0) {
+                    self.webSocket.open(socketAddress)
+                }
+                self.connectionRetries += 1
+            }
+        }
+
+        self.webSocket.event.error = { error in
+            self.log("Error on WebSocket Connection \(error)")
+        }
+
+        // Now actually try to open the WebSocket connection.
+        self.webSocket.open(socketAddress)
+    }
+
+    /// Send the queued logs across the WebSocket connection.
+    public func sendLogDump() {
```

Comment:
I am not sold on the architecture that I implemented, but I wanted to clarify the current separation of concerns and discuss the ramifications of changing the owner of the log-queue.

As it stands now, the separation of concerns is as follows:

- `LogStreamer`: handles capturing logs and timing of sending logs
- `WebSocketMessenger`: handles WebSocket connection, and manages the logs that need to be sent

The reason for putting the WebSocket connection management and log queue management is to provide an abstraction to `LogStreamer` so that it only can enqueue and request for logs to be sent.

Given the current separation of concerns, the `WebSocketMessenger` is not very well named since it does more than provide a WebSocket interface. Possibly something like `LogMessenger` would be better since that removes the connotation that this class only deals with WebSockets.

If we move the log queue to `LogStreamer`, we would also have to move the handling the WebSocket connection there as well, effectively removing the need for `WebSocketMessenger`.

As I said, I am not attached to the current architecture, but I wanted to make sure you knew my logic on this. Also of note, I borrowed some from the architecture of the Android implementation of this library, so it is likely that the way I am doing things is not in the spirit or style of iOS. I'm entirely fine with making this architectural change, but I just wanted to make sure we are on the same page.